content/articles/javascript-screen-recorder/index.md

```diff
@@ -0,0 +1,430 @@
+
+### Getting started
+Our screen recorder application will use full-stack JavaScript. The main libraries on the client-side include:
+- `React`- A frontend JavaScript library for dynamic apps.
+- `socket.io-client` for interacting with the webserver.
+
+For the backend, we will use:
+- `Express.js` - A Node.js framework for servers.
+- `Socket.io` - a Node.js library for developing real-time bi-directional applications using web sockets.
+- `FFmpeg` - an open-source tool for working with multimedia streams such as audio and video.
+
+### Application setup
+Our application structure includes a backend(`server` directory) and a React `client` folder.
+
+First, we will create our application folder and name it `screen-recorder-app`. Then, we will have the two directories inside the folder; the `server` folder for the backend and the `client` folder for the React application.
+
+On your bash terminal, create the `screen-recorder-app` project folder:
+
+```bash
+mkdir screen-recorder-app
+```
+
+Navigate the directory to create the React client using the `create-react-app` utility. Execute the following commands:
+
+```bash
+cd screen-recorder-app
+
+npx create-react-app client
+```
+
+The `create-react-app` command-line tool creates a boilerplate code for our application. However, our entire code will be on the `src/App.js` file.
+
+Our application needs the web sockets interface to reach the backend. For this functionality, let us add the `socket.io-client` and `react-loader-spinner` module:
+
+```bash
+npm install socket.io-client react-loader-spinner
+```
+
+Finally, open the folder on your IDE. For VS Code, run the command:
+
+```bash
+code  .
+```
+
+To set up our component, head over to your `src/App.js` file and import the `useEffect`, `useRef`, and `useState` hooks. Next, from our `socket.io-client` module, import the `io` object to initialize our client.
+
+```js
+import { useEffect, useRef, useState, Fragment } from 'react';
+import { io } from 'socket.io-client';
+
+// adding a simple loading spinner component
+import Loader from 'react-loader-spinner';
+```
+
+On top of our main `App.js` file, declare the app variables that include:
+- Backend local server address as `http://localhost:5000`.
+- The data_chunks of the recorded data and `MediaRecorder` instance interface will provide an API to record `MediaStream`.
+
+```js
+// server address
+const LOCAL_SERVER = 'http://localhost:5000';
+let data_chunks = [];
+
+// MediaRecorder instance
+let media_recorder = null;
+```
+
+We will write everything in the `App.js` file for our client-side React. Let's add a functional component rendered from the JSX with a `<h1>` tag of Recorder App.
+
+```js
+function App() {
+
+// return a JSX of h1
+  return (
+    <Fragment>
+      <h1>Recorder App</h1>
+    </Fragment>
+  )
+}
+```
+
+If we start our server using the CLI command of `npm start`, then head over to our browser, we should see something like:
```

Comment:
I think there is something missing here. Either an image or a code snippet. 
`we should see something like:`???

Please check on this as the article is ready to be published

Reply:
on it, sorry must have missed that.